components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/request/validator/BFSIRequestObjectValidationExtension.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param initialRequestObject request object
+     * @param oAuth2Parameters     oAuth2Parameters
+     * @throws RequestObjectException - RequestObjectException
+     */
+    @Override
+    public boolean validateRequestObject(RequestObject initialRequestObject, OAuth2Parameters oAuth2Parameters)
+            throws RequestObjectException {
+
+        try {
+            if (isRegulatory(oAuth2Parameters)) {
+
+                BFSIRequestObject bfsiRequestObject = new BFSIRequestObject(initialRequestObject);
+
+                Map<String, Object> dataMap = new HashMap<>();
+                final String allowedScopes = getAllowedScopes(oAuth2Parameters);
+                if (StringUtils.isNotBlank(allowedScopes)) {
+                    dataMap.put(IdentityCommonConstants.SCOPE, Arrays.asList(allowedScopes.split(" ")));
+                }
+                // perform BFSI customized validations
+                ValidationResponse validationResponse = obDefaultRequestObjectValidator
+                        .validateBFSIConstraints(bfsiRequestObject, dataMap);
+
+                if (!validationResponse.isValid()) {
+                    // Exception will be caught and converted to auth error by IS at endpoint.
+                    throw new RequestObjectException(RequestObjectException.ERROR_CODE_INVALID_REQUEST,
+                            validationResponse.getViolationMessage());
+                }
+            }
+            return validateIAMConstraints(initialRequestObject, oAuth2Parameters);
+
+        } catch (RequestObjectException e) {
+            log.error("Error while retrieving regulatory property from sp metadata", e);
+            throw new RequestObjectException(RequestObjectException.ERROR_CODE_INVALID_REQUEST, "Error while " +
+                    "retrieving regulatory property from sp metadata");
+        }
+    }
+
+    /**
+     * Validate IAM related logic.
+     * @param requestObject        request object
+     * @param oAuth2Parameters     oAuth2Parameters
+     * @return boolean            true if valid request object
+     * @throws RequestObjectException - RequestObjectException
+     */
+    @Generated(message = "super methods cannot be mocked")
+    boolean validateIAMConstraints(RequestObject requestObject,
+                                   OAuth2Parameters oAuth2Parameters) throws RequestObjectException {
+
+        return super.validateRequestObject(requestObject, oAuth2Parameters);
+    }
+
+
+    /**
+     * Called by validateRequestObject.
+     *
+     * @param requestObject      request object
+     * @param oAuth2Parameters   oAuth2Parameters
+     * @return                true if valid audience
+     */
+    @Generated(message = "Empty method")
+    @Override
+    protected boolean isValidAudience(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) {
+
+        // converted to validation layer
+        return true;
```

Comment:
why are we returning true for all audiences?

Reply:
Because we dont have any custom validation for accelerator level